File: 문제풀이/chap5-7/prob5-7-3.py
# 입력
N = int(input())
A = list(map(int, input().split()))

# 배열A 전체를 정렬
A.sort()

# 답 구하기
Answer = 0
for i in range(1, N + 1):
	Answer += A[i - 1] * (-N + 2 * i - 1)

# 출력
print(Answer)

File: 문제풀이/chap5-7/prob5-7-3.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 입력
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		long[] A = new long[N];
		for (int i = 1; i <= N; i++) {
			A[i - 1] = sc.nextLong();
		}
		
		// 배열A 전체를 정렬
		Arrays.sort(A);
		
		// 답 구하기
		long Answer = 0;
		for (int i = 1; i <= N; i++) {
			Answer += A[i - 1] * (-N + 2L * i - 1L);
		}
		
		// 출력
		System.out.println(Answer);
	}
}